### ISSUE TITLE
BBHN Additional Channels

### ISSUE THREAD

**Issue by ChrisStory** [urlgrey/hsmm-pi#103]
BBHN now has the ability to use channels/frequencies that are unavailable to consumer WiFi devices. 
Would it be possible to modify this project to use these channels (-1 and -2 in BBHN) ?

**Comment by argilo** [comment]
This is something I've been investigating. It's possible, but requires:

* hardware capable of tuning non-standard channels, e.g. Atheros AR9271 chipset
* kernel module patches
* regdb (regulatory database) patches

Lots of useful information is available at http://yo3iiu.ro/blog/?p=1301

I've made the kernel & regdb changes and tuned a TP-Link TL-WN722N to negative channels, but didn't get as far as making HSMM-Pi aware of the negative channels.

My (out-of-date) experiment patching the Raspberry Pi kernel is here: https://github.com/raspberrypi/linux/compare/rpi-3.18.y...argilo:rpi-3.18.y-ath9k-ham

**Comment by argilo** [comment]
Of course, recompiling the kernel during HSMM-Pi install would take a ridiculous amount of time, so it would be best to distribute HSMM-Pi as a flashable SD card image if we make a version capable of using negative channels.

**Comment by mathisono** [comment]
Your post was great to read... I wasn't completely confident in undertaking
the mod. Ive had the TL-WN722N for a while and love the card... I had some
power consumption problems on the first 2 generations of RPi B. They went
away with the Rpi B with 512mb. My HSMM-pi node is the last thing on ch1.

Do you think you can swing a scripting the kernel and regdb patches. It
should not be part of the the standard build, it could a config flag. Have
the install script ask.

73 Mathison kj6dzb

**Comment by argilo** [comment]
@mathisono The blog post isn't mine; it was written by @BogdanDIA. I just followed his instructions.

**Comment by sv1** [comment]
Here is a nice "how-to" on how you can compile only one module (ex. the wifi card driver) without compiling the whole kernel

http://askubuntu.com/questions/515407/how-recipe-to-build-only-one-kernel-module

**Comment by virtualdxs** [comment]
What's the status on this?

**Comment by raysuelzer** [comment]
This is a good idea. I have been poking around a bit in the various drivers source code, the most difficult part of having this integrate with the non part97 AREDN nodes seems like it will be the fact that most of them transmit at 10mhz or 5mhz, and all of the drivers expect either 20MHzor 40MHzbandwidth.

**Comment by KB3RRL** [comment]
Has any progress been made on this excellent idea to update HSMM-Pi to permit it use on the -1 and -2 channels in the 2.4GHz band with a bandwidth of 5, 10, or 20MHz?  This would serve as a useful device for mobile deployment.
